node/lib/package/LocalPackageSource.ts

```diff
@@ -330,9 +363,61 @@ export class LocalPackageSource extends PackageSource {
     }
 
     private async loadOnboardingPackageInfoAsync(
-            onboardingName: string): Promise<PackageInfo | null> {
-        // TODO: Load onboarding package info from XML + package.json
-        return Promise.resolve(null);
+        onboardingName: string): Promise<PackageInfo | null> {
+
+        // By convention there is package.json files under 'js' directory. 
+        let onboardingManifestPath = onboardingName + "/js/manifest.xml";
+        let onboardinPackageJsonPath = onboardingName + "/js/package.json";
+        let translatorModulePath = onboardingName + "/js/thingOnboarding";
+
+        if (!(await fs.exists(path.join(this.sourceDirectory, translatorModulePath + ".js"))) ||
+            !(await fs.exists(path.join(this.sourceDirectory, onboardingManifestPath))) ||
+            !(await fs.exists(path.join(this.sourceDirectory, onboardinPackageJsonPath)))) {
+            // The requested onboarding package was not found.
+            return null;
+        }
+
+        let packageJson: any = await this.loadJsonAsync(onboardinPackageJsonPath);
+        let manifestXml: any = await this.loadXmlAsync(onboardingManifestPath);
+        let manifestXmlRoot: any = manifestXml && manifestXml.manifest;
+
+        if (!packageJson || !manifestXmlRoot) {
+            return null;
+        }
+
+        // Parse schema info from the manifest.
+        let schemaInfos: any[] = [];
+        let schemaModulePaths: string[] = [];
+        if (Array.isArray(manifestXmlRoot.schemas) &&
+                manifestXmlRoot.schemas.length == 1 &&
+                Array.isArray(manifestXmlRoot.schemas[0].schema)) {
+            let schemaElements = manifestXmlRoot.schemas[0].schema;
+           // TODO: Will onboarding schemas also have 'main' schema concept ?
+
+            schemaElements.forEach((schemaElement: any) => {
+                let schemaId: string = schemaElement.$.id;
+                if (schemaId) {
+                    // By convention the schema module file and its parent directory
+                    // both match the name of the schema.
+                    let schemaModulePath: string = packageJson.name + "/" + schemaId;
+                    schemaModulePaths.push(schemaModulePath);
```

Comment:
This should be packageJson.name + "/" + schemaId + "/" + schemaId, because the schemaId is the name of both the directory and the file.